src/strategies/path-balance-staked-and-locked/index.ts

```diff
@@ -0,0 +1,95 @@
+import { BigNumberish } from '@ethersproject/bignumber';
+import { formatUnits } from '@ethersproject/units';
+import { Multicaller } from '../../utils';
+import { multicall } from '../../utils';
+
+export const author = 'bonustrack';
+export const version = '0.1.1';
+
+const tokenAbi = [
+  'function balanceOf(address account) external view returns (uint256)'
+];
+
+const stakingAbi = [
+  'function balanceOf(address account) external view returns (uint256)'
+];
+
+function chunk(array, chunkSize) {
+  const tempArray: any[] = [];
+  for (let i = 0, len = array.length; i < len; i += chunkSize)
+    tempArray.push(array.slice(i, i + chunkSize));
+  return tempArray;
+}
+
+export async function strategy(
+  space,
+  network,
+  provider,
+  addresses,
+  options,
+  snapshot
+): Promise<Record<string, number>> {
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+
+  const stakingPool = new Multicaller(network, provider, stakingAbi, { blockTag });
+  const tokenPool = new Multicaller(network, provider, tokenAbi, { blockTag });
+  
+  addresses.forEach((address) => {
+    stakingPool.call(address, options.stakingAddress, 'balanceOf', [address]);
+    tokenPool.call(address, options.tokenAddress, 'balanceOf', [address]);
+ });
+
+ const [stakingResponse, tokenResponse] : [
+  Record<string, BigNumberish>,
+  Record<string, BigNumberish>
+ ] = await Promise.all([stakingPool.execute(), tokenPool.execute()]);
+
+ let callData: [any, string, [any]][] = [];
+  addresses.map((address: any) => {
+    options.lockedAddresses.map((lockedAddress: any) => {
+      callData.push([lockedAddress, options.methodABI.name, [address]]);
+    });
+  });
+
+  callData = [...chunk(callData, 2000)]; // chunking the callData into multiple arrays of 2000 requests
+  let response: any[] = [];
+  for (let i = 0; i < callData.length; i++) {
+    const tempArray = await multicall(
+      network,
+      provider,
+      [options.methodABI],
+      callData[i],
+      { blockTag }
+    );
+    response.push(...tempArray);
+  }
```

Comment:
Something is not correct here. If the callData array is big ( in the case of a big number of voters) each address will send a separate request, which doesn't scale well. can you see if there is an alternate way to achieve this?

Reply:
thanks. let me try to modify this

Reply:
i think the new version should achieve better scaling by just using multicalls for the different contracts